### PR TITLE
Ignore value option for metricdatatest

### PR DESCRIPTION
**Pull request by RangelReale**
Follow-up from [this PR](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/3769#discussion_r1289773894) on the contrib repo.

Duration metrics are not predictable, so it was suggested to add an option in `metricdatatest` to allow ignoring values when comparing metrics.

### CODE REVIEW

**Review by MadVikingGod — APPROVED** [review]

**Review by hanyuancheung — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

File: sdk/metric/metricdata/metricdatatest/comparisons.go [hunk 1]
```diff
@@ -518,14 +523,16 @@ func equalExemplars[N int64 | float64](a, b metricdata.Exemplar[N], cfg config)
...
+	if !cfg.ignoreValue {
+		if a.Value != b.Value {
+			reasons = append(reasons, notEqualStr("Value", a.Value, b.Value))
+		}
+		if !equalSlices(a.SpanID, b.SpanID) {
+			reasons = append(reasons, notEqualStr("SpanID", a.SpanID, b.SpanID))
+		}
+		if !equalSlices(a.TraceID, b.TraceID) {
+			reasons = append(reasons, notEqualStr("TraceID", a.TraceID, b.TraceID))
+		}
```
Comment (MrAlias) [inline sdk/metric/metricdata/metricdatatest/comparisons.go:535]: Should the trace information be ignored here? It doesn't seem like the value.

Comment (RangelReale) [reply sdk/metric/metricdata/metricdatatest/comparisons.go]: You're right, fixed.

**Review by pellared — COMMENTED** [review]
Can you please add an entry to `CHANGELOG.md`?

**Comment by RangelReale** [comment]
> Can you please add an entry to `CHANGELOG.md`?

done.

**Review by pellared — APPROVED** [review]

**Pull request merged**